src/reducers/linter.tsx

```diff
@@ -85,3 +105,108 @@ export const getMessageMap = (result: ExternalLinterResult) => {
 
   return msgMap;
 };
+
+export type LinterState = {
+  forVersionId: undefined | number;
+  isLoading: boolean;
+  messageMap: undefined | LinterMessageMap;
+};
+
+export const initialState: LinterState = {
+  forVersionId: undefined,
+  isLoading: false,
+  messageMap: undefined,
```

Comment:
Why is this reducer storing data for at most one version?

Reply:
I couldn't think of any realistic situation where we'll need to load multiple versions on the same page. Limiting the reducer to one version keeps the structure simple and easy to work with. The selector will return undefined for version mismatches which will trigger a refresh of the data.

Reply:
Like everything else: we load the data only once. If the user goes back and forth on different files, we don't call the API all the time and it loads faster.

Reply:
> we don't call the API all the time

I discovered (~2h ago) that this got lost somewhere... on the browse page. We re-fetch/load the data all the time, but because data are stored already, it feels very fast and I suppose that's why no one noticed it before. We just do an extra API call every time /o\

Reply:
Re-fetching data isn't a bad thing because the browser will serve a cached API response and that is very fast. But, sure, I guess I could change it to preserve all fetched versions to accommodate quick-switching between versions.

Reply:
> maybe we could start with your single-version state and see how it goes in real life?

As discussed on IRC ^ we're going to try it as is. It will probably be fast enough because of how the browser will cache the API response.

Reply:
Given that this works already, caching other linter results could be done as a follow-up (if we needed it).